lib/src/bottom_sheet.dart

```diff
@@ -231,7 +231,8 @@ class _ModalBottomSheetState extends State<ModalBottomSheet>
   void _handleScrollUpdate(ScrollNotification notification) {
     if (notification.metrics.pixels <= notification.metrics.minScrollExtent) {
       //Check if listener is same from scrollController
-      if (_scrollController.position.pixels != notification.metrics.pixels) {
+      if (_scrollController.hasClients &&
```

Comment:
I wouldn't change the widget tree either.

Currently did not get what would not work with my change, but I added your line to the code.

Reply:
Because _handleScrollUpdate should only work when _scrollController is the one scrolling. 
That is why i use this  if (_scrollController.position.pixels != notification.metrics.pixels)
If the position of the scroller i no the one of the notification return and don't continue.

If you add _scrollController.hasClients there, the scrollController is not attached and there is another one scrolling: it would still call the code below